test/test-recipes.js

```diff
@@ -56,13 +57,55 @@ test('recipes with supergrammars', function(t) {
 });
 
 test('recipes involving parameterized rules', function(t) {
-  var g = ohm.grammar(
-      'G {\n' +
-      '  foo = bar<"0", "1">\n' +
-      '  bar<x, y> = "a" x -- one\n' +
-      '            | "b" y -- two\n' +
-      '}');
+  var g = testUtil.makeGrammar([
+    'G {',
+    '  foo = bar<"0", "1">',
+    '  bar<x, y> = "a" x -- one',
+    '            | "b" y -- two',
+    '}'
+  ]);
   t.ok(toGrammar(g.toRecipe()).match('a0').succeeded());
   t.ok(toGrammar(g.toRecipe()).match('b1').succeeded());
   t.end();
 });
+
+test('recipes with source', function(t) {
+  var ns = testUtil.makeGrammars([
+    ' G {',  // Deliberately start with leading space.
+    '  Start = ident*',
+    '  ident = letter /* foo */ identPart*',
+    '  identPart = alnum',
+    '}',
+    ' G2 <: G {',
+    '  Start := (ident | number)*',
+    '  identPart += "$"',
+    '  number = digit+',
+    '}'
+  ]);
+  var g = toGrammar(ns.G.toRecipe());
+  t.equal(g.ruleBodies.Start.interval.contents, 'ident*');
+  t.equal(g.ruleBodies.ident.interval.contents, 'letter /* foo */ identPart*');
+
+  // Try re-parsing the grammar based on the source retained in the recipe.
+  var reconsitutedGrammar = ohm.grammar(g.definitionInterval.contents);
+  t.equal(reconsitutedGrammar.toRecipe(), ns.G.toRecipe());
+
+  var g2 = toGrammar(ns.G2.toRecipe());
+  t.equal(g2.ruleBodies.Start.interval.contents, '(ident | number)*', 'overridden rule');
+
+  // TODO: Figure out how to describe the interval for extended rules.
```

Comment:
Shouldn't it just be the interval of the extended rule's declaration, i.e.,

  ruleName += ...

?

Were you thinking that it should also include the supergrammar's version of this rule? IMO we should be able to get to that, but it shouldn't be in the interval of this rule.